### ISSUE TITLE
[Bug]: Numerica Edit's do not allow a zero after a decimal

### ISSUE THREAD

**Issue by Eltee-Taiwo** [Megabit/Blazorise#5651]
### Blazorise Version

1.5.2

### What Blazorise provider are you running on?

Bootstrap5

### Link to minimal reproduction or a simple code snippet

I figured the demo site would be the best example to use to rule out potential issues with my code: https://bootstrap5demo.blazorise.com/tests/forms

### Steps to reproduce

1. Scroll down to Number edit field
2. Attempt to enter in the value 1.01

### What is expected?

The Input field takes the value 1.01

### What is actually happening?

Pressing the zero seems to work like a backspace and instead removes the decimal so you end up with 11

### What browsers do you see the problem on?

Chrome

### Any additional comments?

I tested the Tailwind demo site as well to rule out if it was issue with the framework

**Comment by Eltee-Taiwo** [comment]
If javascript is disabled, then the error doesn't occur. Points to a javascript file causing the issue.

**Comment by stsrki** [comment]
> Points to a javascript file causing the issue.

Can you explain what JS file in particular is causing the problem? Thanks.

**Comment by David-Moreira** [comment]
I tried typing what OP suggested, as soon as I write **1.0** it turns to **1**. So I can't really add additional decimals, when the first decimal is 0. Not sure if intended.
However If I type, 1.1111 I can write it just fine.

**Comment by stsrki** [comment]
I think this is an old and known problem with the browser. Since we are using native HTML `input type="number"` as soon as Blazor submits the value back to browser the browser strips the zero behind the decimal, and round to the whole number. I really don't know how to fix this since we don't have any way of controlling `input type="number"` or knowing if the value is partial or not.

**Comment by Eltee-Taiwo** [comment]
> I think this is an old and known problem with the browser. Since we are using native HTML `input type="number"` as soon as Blazor submits the value back to browser the browser strips the zero behind the decimal, and round to the whole number. I really don't know how to fix this since we don't have any way of controlling `input type="number"` or knowing if the value is partial or not.

@stsrki  I'm not quite sure if that's accurate. 

When I test the `input type ="number"` on the [mozilla dev site](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/number), I am unable to recreate this issue.

.
.
.


> > Points to a javascript file causing the issue.
> 
> Can you explain what JS file in particular is causing the problem? Thanks.

Unfortunately I haven't been able to pin down which JS file is causing it, but I do know that if JS is disabled after the page has loaded on the blazorise demo site, I cannot recreate the issue.

**Comment by stsrki** [comment]
That's because there is no `middleman` in native HTML. It is fully controlled by the browser. Now, when you put Blazor in between, you get the following flow

1. The user writes the value
2. value is sent to Blazor
3. Blazor raises onchange/ininpuit event
4. we need to parse it
5. if properly parsed, then update the internal state
6. update input value with internal state value

The parsing part and the updating part is critical. We could control it with `preventDefault` if we detect the partial value, but unlike with regular JS, the `preventDefault` will always be one event behind. In Blazor all events are asynchronous, while `preventDefault` is synchronous so we cannot use that. We have tried it without any luck.

Now, this is not only Blazor problem. Every SPA framework can have similar problem, (https://github.com/facebook/react/issues/9402) but since they are mostly JS it is somewhat easier for them to fix it.

My advice is to use NumericEdit for very simple numbers, and use NumericPicker for everything else. That's why we have created it in the first place.
